write-fonts/src/tables/gvar.rs

```diff
@@ -457,19 +474,37 @@ struct GlyphDataWriter<'a> {
 
 impl FontWrite for GlyphDataWriter<'_> {
     fn write_into(&self, writer: &mut TableWriter) {
-        assert!(self.long_offsets, "short offset logic not implemented");
-        let mut last = 0u32;
-        last.write_into(writer);
+        if self.long_offsets {
+            let mut last = 0u32;
+            last.write_into(writer);
 
-        // write all the offsets
-        for glyph in self.data {
-            last += glyph.compute_size();
+            // write all the offsets
+            for glyph in self.data {
+                last += glyph.compute_size();
+                last.write_into(writer);
+            }
+        } else {
+            // for short offsets we divide the real offset by two; this means
+            // we will have to add padding if necessary
+            let mut last = 0u16;
             last.write_into(writer);
+
+            // write all the offsets
+            for glyph in self.data {
+                let size = glyph.compute_size();
+                // ensure we're always rounding up to the next 2
+                let short_size = (size / 2) + size % 2;
+                last += short_size as u16;
+                last.write_into(writer);
+            }
         }
         // then write the actual data
         for glyph in self.data {
             if !glyph.is_empty() {
                 glyph.write_into(writer);
+                if !self.long_offsets && glyph.length % 2 != 0 {
+                    writer.pad_to_2byte_aligned();
+                }
```

Comment:
You dont' need && glyph.length % 2 here as pad_to_2byte_aligned will happily nop if htat's true?